tools/autograd/gen_variable_type.py

```diff
@@ -370,7 +370,8 @@
 
 # Generate a file that lists all functions and their schema string. Used for XLA
 REGISTRATION_DECLARATION = CodeTemplate("""\
-${return_type} ${api_name}(${declaration_formals}); // {"schema": "${schema_string}", "compound": "${compound}"}
+${return_type} ${api_name}(${declaration_formals}); \
+// {"schema": "${schema_string}", "compound": "${compound}", "has_math_kernel": "${has_math_kernel}"}
```

Comment:
Wondering if we can just port this codegen straight to the new codegen, then we don't have to expose `has_math_kernel` in Declarations.yaml.

Reply:
Unfortunately it has to stay in autograd codegen for now since we need access to `whether an op has derivative` to cover DispatchSub ops. Whether the kernel comes from disptach stub is not available anywhere in codegen and can only be approximated by the fact that almost all dispatch stub ops have derivatives). So we need to add this information in native_functions.yaml before we could move this codegen. More details can be found at: https://fb.quip.com/1B61AfNcuozU

Reply:
OK. We'll be out of this situation soon, but this is fine for this PR.